### PR TITLE
[FLINK-19655][flink-table-runtime-blink]  add super.open() and write unit test for temporal process join

### PR DESCRIPTION
**Pull request by seunjjs**
## What is the purpose of the change

The ultimate goal of this PR is to fix a bug where NPE is thrown when using blink planner and TemporalTableFunction after setting IdleStateRetentionTime .TemporalProcessTimeJoinOperator should initialize timer service for processing time.
See #FLINK-19655

## Brief change log
  - Add super.open() in LegacyTemporalProcessTimeJoinOperator#open() to solve NPE
  - Write two unit test for LegacyTemporalProcessTimeJoinOperator to verify the code worked fine

## Verifying this change

This change added tests and can be verified as follows:
  - Added test that validates that:
    - 1.test proctime temporal join when non state clear. 
    - 2.test proctime temporal join when set state clear.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`:  no
  - The serializers:   no 
  - The runtime per-record code paths (performance sensitive):  no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper:  no 
  - The S3 file system connector: no 

## Documentation

  - Does this pull request introduce a new feature?  no
  - If yes, how is the feature documented?  not documented

### CODE REVIEW

File: flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/runtime/operators/join/temporal/LegacyTemporalProcessTimeJoinOperatorTest.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+import org.apache.flink.table.types.logical.VarCharType;
+
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.flink.table.runtime.util.StreamRecordUtils.insertRecord;
+
+
```
Comment (leonardBang) [inline flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/runtime/operators/join/temporal/LegacyTemporalProcessTimeJoinOperatorTest.java:38]: single blank line

File: flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/join/temporal/LegacyTemporalProcessTimeJoinOperator.java [hunk 2]
```diff
@@ -73,6 +73,7 @@ public void open() throws Exception {
 		this.outRow = new JoinedRowData();
 		// consider watermark from left stream only.
 		super.processWatermark2(Watermark.MAX_WATERMARK);
+		super.open();
```
Comment (leonardBang) [inline flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/join/temporal/LegacyTemporalProcessTimeJoinOperator.java:76]: move function call of super to the beginning

File: flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/runtime/operators/join/temporal/LegacyTemporalProcessTimeJoinOperatorTest.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+	/** test proctime temporal join when set state clear. **/
+	@Test
+	public void testStateCleanUp() throws Exception {
+		final int minRetentionTime = 10;
+		LegacyTemporalProcessTimeJoinOperator joinOperator = new LegacyTemporalProcessTimeJoinOperator(
+				rowType,
+				joinCondition,
+				minRetentionTime,
+				minRetentionTime * 3 / 2);
```
Comment (leonardBang) [inline flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/runtime/operators/join/temporal/LegacyTemporalProcessTimeJoinOperatorTest.java:116]: `final int maxRententionTime = minRetentionTime * 3 / 2`
we can reuse  `maxRententionTime` in following code

File: flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/join/temporal/LegacyTemporalProcessTimeJoinOperator.java (hunk 2 above)
Comment (leonardBang) [reply flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/join/temporal/LegacyTemporalProcessTimeJoinOperator.java:76]: and I believe `TemporalRowTimeJoinOperator.java` exists same problem, could you help fix?

File: flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/runtime/operators/join/temporal/LegacyTemporalProcessTimeJoinOperatorTest.java [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+			throws Exception {
+
+		return new KeyedTwoInputStreamOperatorTestHarness<>(
+				temporalJoinOperator,
+				keySelector,
+				keySelector,
+				keyType);
+	}
+
+}
```
Comment (leonardBang) [inline flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/runtime/operators/join/temporal/LegacyTemporalProcessTimeJoinOperatorTest.java:150]: checkstyle: one blank line at the end of file

**Review by leonardBang — CHANGES_REQUESTED** [review]
@seunjjs Thanks for the contribution, I left some comments.

File: flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/runtime/operators/join/temporal/LegacyTemporalProcessTimeJoinOperatorTest.java (hunk 1 above)
Comment (seunjjs) [reply flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/runtime/operators/join/temporal/LegacyTemporalProcessTimeJoinOperatorTest.java]: fixed.

File: flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/runtime/operators/join/temporal/LegacyTemporalProcessTimeJoinOperatorTest.java (hunk 3 above)
Comment (seunjjs) [reply flink-table/flink-table-runtime-blink/src/test/java/org/apache/flink/table/runtime/operators/join/temporal/LegacyTemporalProcessTimeJoinOperatorTest.java]: fixed.

File: flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/join/temporal/LegacyTemporalProcessTimeJoinOperator.java (hunk 2 above)
Comment (seunjjs) [reply flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/join/temporal/LegacyTemporalProcessTimeJoinOperator.java]: fixed.

**Comment by leonardBang** [comment]
@seunjjs Thanks for the update, the code looks fine, but the code check style is a little missy maybe you're using windows system.
I add a commit to help format that, you can cherry-pick this https://github.com/leonardBang/flink/commit/cd333bf537031e7ff5049b3a59da38d7859f416b directly.

**Comment by seunjjs** [comment]
Thank you. My system is win10, and I have followed this guide [https://ci.apache.org/projects/flink/flink-docs-release-1.11/flinkDev/ide_setup.html#checkstyle-for-java](url) to set Checkstyle-IDEA plugin. But if I run ctrl+alt+l, then the code style is different from yours. Do you know how can I get the right checkstyle plugin config on windows?😥

**Comment by leonardBang** [comment]
> Thank you. My system is win10, and I have followed this guide [https://ci.apache.org/projects/flink/flink-docs-release-1.11/flinkDev/ide_setup.html#checkstyle-for-java](url) to set Checkstyle-IDEA plugin. But if I run ctrl+alt+l, then the code style is different from yours. Do you know how can I get the right checkstyle plugin config on windows?😥

I also followed this guide, I'm using mac, I think you've set right check style, the only difference between us is using single tab or double tabs before a new line, both the two formats exits in Flink codebase and can pass the code style check well. 
I also improve a little about the note, parameter name and redundant blank lines, that's why I add the format commit.

**Review by leonardBang — APPROVED** [review]
LGTM

**Review by wuchong — APPROVED** [review]
LGTM.

**Pull request merged**